### PR TITLE
EZP-28831: Draft conflict window after clicking edit button of existing draft in Dashboard and Sub-items list

### PR DESCRIPTION
**Pull request by mikadamczyk**
| Question      | Answer
| ------------- | ---
| Tickets       | https://jira.ez.no/browse/EZP-28831
| Bug fix?      |no
| New feature?  | yes
| BC breaks?    |no
| Tests pass?   | yes
| Doc needed?   | no
| License       | [GPL-2.0](https://github.com/ezsystems/ezplatform-admin-ui/blob/master/LICENSE)

Draft conflict window after clicking edit button of existing draft in Dashboard, Sub-items list, Search results list and Content item using this URL in Link Manager.

screen shot 2018-02-12 at 2 51 21 pm

screen shot 2018-02-26 at 11 56 31 am

screen shot 2018-02-26 at 11 56 09 am

#### Checklist:
- [x] For Edit button in Dashboard
- [x] For Edit button on Sub-items list
- [x] For Edit button in Search results list
- [x] For Edit button in Content item using this URL in Link Manager
- [x] Coding standards (`$ composer fix-cs`)
- [x] Ready for Code Review

### CODE REVIEW

File: src/bundle/Resources/public/js/scripts/admin.location.view.js [hunk 1]
```diff
@@ -19,12 +19,35 @@
...
+        const submitVersionEditForm = () => {
+            doc.querySelector('#form_subitems_content_edit_content_info').value = contentId;
+            doc.querySelector('#form_subitems_content_edit_version_info_content_info').value = contentId;
+            doc.querySelector('#form_subitems_content_edit_version_info_version_no').value = content.CurrentVersion.Version.VersionInfo.versionNo;
+            doc.querySelector(`#form_subitems_content_edit_language_${content.mainLanguageCode}`).checked = true;
+            doc.querySelector('#form_subitems_content_edit_create').click();
+        };
+        const addDraft = () => {
+            submitVersionEditForm();
+            $('#version-draft-conflict-modal').modal('hide');
```
Comment (sunpietro) [inline src/bundle/Resources/public/js/scripts/admin.location.view.js]: This part sould be cached in a variable, like `const modal = $('#version-draft-conflict-modal');` in the higher scope. Then it should be reused in functions like `addDraft`.

File: src/bundle/Resources/public/js/scripts/admin.location.view.js [hunk 2]
```diff
@@ -19,12 +19,35 @@
...
+        };
+        const showModal = (modalHtml) => {
+            const wrapper = doc.querySelector('.ez-modal-wrapper');
+            wrapper.innerHTML = modalHtml;
```
Comment (sunpietro) [inline src/bundle/Resources/public/js/scripts/admin.location.view.js]: CS: please an empty line above for better readability.

File: src/bundle/Resources/public/js/scripts/admin.location.view.js [hunk 3]
```diff
@@ -19,12 +19,35 @@
...
+            wrapper.querySelector('.ez-btn--add-draft').addEventListener('click', addDraft, false);
+            [...wrapper.querySelectorAll('.ez-btn--prevented')].forEach(btn => btn.addEventListener('click', event => event.preventDefault(), false));
+            $('#version-draft-conflict-modal').modal('show');
+        };
+        fetch(checkVersionDraftLink, {
+            credentials: 'same-origin'
+        }).then(function (response) {
+            if (response.status === 409) {
```
Comment (sunpietro) [inline src/bundle/Resources/public/js/scripts/admin.location.view.js]: I believe some sort of comments could improve code readability.

**Comment by lserwatka** [comment]
@micszo who is on this one?

**Comment by micszo** [comment]
@barbaragr already started clarifying issues around this topic.

**Comment by micszo** [comment]
I believe there was a decision to split it into two tickets, correct @mikadamczyk ?

**Comment by mikadamczyk** [comment]
I hope so. I talked to Sylvain yesterday. It's about adding the possibility of choosing a language to Edit button. It seems to me that it will be in the new task.

**Pull request merged**